jsx/Form.js

```diff
@@ -294,72 +303,38 @@ SearchableDropdown.defaultProps = {
  * Select Component
  * React wrapper for a simple or 'multiple' <select> element.
  */
-var SelectElement = React.createClass({
-
-  propTypes: {
-    name: React.PropTypes.string.isRequired,
-    options: React.PropTypes.object.isRequired,
-    label: React.PropTypes.string,
-    value: React.PropTypes.oneOfType([
-      React.PropTypes.string,
-      React.PropTypes.array
-    ]),
-    id: React.PropTypes.string,
-    class: React.PropTypes.string,
-    multiple: React.PropTypes.bool,
-    disabled: React.PropTypes.bool,
-    required: React.PropTypes.bool,
-    emptyOption: React.PropTypes.bool,
-    hasError: React.PropTypes.bool,
-    errorMessage: React.PropTypes.string,
-    onUserInput: React.PropTypes.func
-  },
-
-  getDefaultProps: function() {
-    return {
-      name: '',
-      options: {},
-      label: '',
-      value: undefined,
-      id: '',
-      class: '',
-      multiple: false,
-      disabled: false,
-      required: false,
-      emptyOption: true,
-      hasError: false,
-      errorMessage: 'The field is required!',
-      onUserInput: function() {
-        console.warn('onUserInput() callback is not set');
-      }
-    };
-  },
+class SelectElement extends Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+  }
 
-  handleChange: function(e) {
-    var value = e.target.value;
-    var options = e.target.options;
+  handleChange(e) {
+    let value = e.target.value;
+    let options = e.target.options;
 
     // Multiple values
     if (this.props.multiple && options.length > 1) {
       value = [];
-      for (var i = 0, l = options.length; i < l; i++) {
+      for (let i = 0, l = options.length; i < l; i++) {
```

Comment:
It is good practice not to use the length property in the loop definition because it will be accessed at each iteration. 
Further more, line 317 to line 324 ciould be change for a map function that always return an array. (empty if options.length = 0, whit 1 value if options.length = 1 ) and it shouldm be fine for the select value anyway

Reply:
Okay, I'll look at refactoring this in my next commit.